sdk/trace/span.go

```diff
@@ -189,32 +189,6 @@ func (s *span) SetName(name string) {
 	makeSamplingDecision(data)
 }
 
-// AddLink implements Span interface. Specified link is added to the span.
-// If the total number of links associated with the span exceeds the limit
-// then the oldest link is removed to create space for the link being added.
-func (s *span) AddLink(link apitrace.Link) {
-	if !s.IsRecording() {
-		return
-	}
```

Comment:
Move this verification to the `addLink` method (before locking the mutex) to not change the behavior of the method.